### DISCUSSION TITLE
Pull request merge options

### DISCUSSION THREAD

**Discussion by gasrodriguez** [community/community#19728] [Pull Requests]
Hi all. Currently there are three flavors of PR merge:
![image](https://user-images.githubusercontent.com/25135385/176558966-99209fd2-33c8-4692-b196-256f2be5dcbe.png)

In my current flow I want to have only one commit in the trunk branch (no a merge commit), with the PR # added to the commit message.
For this reason I never use `Create a merge commit`. 
Almost always use `Squash and merge`, but if someone else pushed to the trunk branch before, I had to do a manual rebase to get the expected result, otherwise it will create a merge commit. 
Sometimes I could just do a `Rebase and merge` because my branch was already squashed, but for some reason `Rebase and merge` doe _not_ append the PR # to the commit message. Why is that? 
What would fit my workflow best is an option to do `Squash, rebase and merge` all in a single action.
If this is too hard, then the second best will be to append the PR # or allow editing the commit message on `Rebase and merge`, so will get the desired result without manual operations in more cases than now.

**Comment by thuongtruong1009** [comment]
Hi @gasrodriguez 
The first option is to create and merge commit: you commit everything in a new branch, after it is done, you will merge it into the main branch, and that action is recorded in history through merge PR
The second option is a squash and merge: if there are many commits in the new branch, they will merge all the commits into a single final commit, then merge them into the main branch. And in history, only 1 commit is recorded for this action
The 3rd option is rebase and merge: you will merge the new branch directly into the main branch and history will not record it through PR.

Hope useful for you! 🥰

**Comment by gasrodriguez** [comment]
Hi @thuongtruong1009 thanks for the reply. What you explain is how it works now and I'm aware of it, but I don't want one or more commits in the branch plus a "merge commit" in the trunk branch. I want what you describe for Rebase & merge "you will merge the new branch directly into the main branch" but with a squash so instead of N commits I get just 1, and I want the commit message of this single commit having the PR # appended, which doesn't happen with Rebase & merge. So AFAIK this is not possible with none of the three options, so this a **feature request.**
